### ISSUE TITLE
`indent`: multiline TS generic indent is not correctly managed

### ISSUE THREAD

**Issue by paztis** [eslint-stylistic/eslint-stylistic#262]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

With classic @stylistic/indent configuration to 2 spaces, Generics are not correctly managing the multiline indentation

```
type TGeneric<A, B, C, D> = void;

type TTest = TGeneric<
  string,
  string,
  string,
  string
>;

```

produce this error

ESLint: Expected indentation of 0 spaces but found 2.(@stylistic/indent)

I can't found any configuration to support this indentation

### Reproduction

https://stackblitz.com/github/eslint-stylistic/repro-template?file=src%2Ftest.ts,package.json

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by paztis** [comment]
only valid workaround is to add TSTypeParameterInstantiation to ignoredNodes option.
But in this case it accepts any indentation, not only 2

**Comment by laverdet** [comment]
You also need to ignore "TSTypeAnnotation" to catch this case:

```ts
function foo(): NonNullable<
  string
> {
  return "";
}
```

**Comment by 9romise** [comment]
Has been fixed by #256.
